### PR TITLE
Added Image Dataset Generation And Consolidated synthesis branch

### PR DESCRIPTION
**Pull request by Ensify**

### CODE REVIEW

File: doctr/datasets/vocabs.py
Comment (kasuba-badri-vishal) [inline doctr/datasets/vocabs.py:1]: No need to change this vocab part I believe

File: .gitignore
Comment (kasuba-badri-vishal) [inline .gitignore:1]: Any data like fonts, sample_data or samples should be present outside the repo, so no need to add it here, the path could be specified accordingly. Why is this added in .gitignore, because all wont use it. Not a good practise

File: misc_code/analyse.py
Comment (kasuba-badri-vishal) [inline misc_code/analyse.py:1]: What is the utility of analyse.py python file

File: misc_code/infer.py
Comment (kasuba-badri-vishal) [inline misc_code/infer.py:1]: Inference code could be added in separate commit, also this is not required for now with the current work

File: misc_code/preprocess.py
Comment (kasuba-badri-vishal) [inline misc_code/preprocess.py:1]: Specify the code as text_preprocess and mention it properly in code

File: references/recognition/train_pytorch.py
Comment (kasuba-badri-vishal) [inline references/recognition/train_pytorch.py:1]: Since this is respective to our version of training for synth data, It is better to create train_pytorch_synth.py file and use it, instead of changing original train_pytorch.py

**Review by kasuba-badri-vishal — CHANGES_REQUESTED** [review]

**Review by Ensify — COMMENTED** [review]
All Suggested Changes Done
